diff --git a/node_modules/@solana-mobile/mobile-wallet-adapter-protocol/lib/esm/index.browser.js b/node_modules/@solana-mobile/mobile-wallet-adapter-protocol/lib/esm/index.browser.js
index e001c2e..ef3feb3 100644
--- a/node_modules/@solana-mobile/mobile-wallet-adapter-protocol/lib/esm/index.browser.js
+++ b/node_modules/@solana-mobile/mobile-wallet-adapter-protocol/lib/esm/index.browser.js
@@ -531,6 +531,9 @@ function launchAssociation(associationUrl) {
             // is installed or load the actual target webpage if not.
             window.location.assign(associationUrl);
         }
+        else if ('Telegram' in window) {
+          window.open(associationUrl);
+        }
         else {
             // The association URL has a custom protocol (eg. `solana-wallet:`)
             try {
diff --git a/node_modules/@solana-mobile/mobile-wallet-adapter-protocol/lib/esm/index.js b/node_modules/@solana-mobile/mobile-wallet-adapter-protocol/lib/esm/index.js
index e001c2e..ef3feb3 100644
--- a/node_modules/@solana-mobile/mobile-wallet-adapter-protocol/lib/esm/index.js
+++ b/node_modules/@solana-mobile/mobile-wallet-adapter-protocol/lib/esm/index.js
@@ -531,6 +531,9 @@ function launchAssociation(associationUrl) {
             // is installed or load the actual target webpage if not.
             window.location.assign(associationUrl);
         }
+        else if ('Telegram' in window) {
+          window.open(associationUrl);
+        }
         else {
             // The association URL has a custom protocol (eg. `solana-wallet:`)
             try {
