diff --git a/node_modules/@solana/wallet-adapter-react/lib/cjs/getEnvironment.js b/node_modules/@solana/wallet-adapter-react/lib/cjs/getEnvironment.js
index 73b6706..5c4a1fd 100644
--- a/node_modules/@solana/wallet-adapter-react/lib/cjs/getEnvironment.js
+++ b/node_modules/@solana/wallet-adapter-react/lib/cjs/getEnvironment.js
@@ -9,7 +9,7 @@ var Environment;
     Environment[Environment["MOBILE_WEB"] = 1] = "MOBILE_WEB";
 })(Environment = exports.Environment || (exports.Environment = {}));
 function isWebView(userAgentString) {
-    return /(WebView|Version\/.+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+)|; wv\).+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+))/i.test(userAgentString);
+    return /(Telegram|WebView|Version\/.+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+)|; wv\).+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+))/i.test(userAgentString);
 }
 function getEnvironment({ adapters, userAgentString }) {
     if (adapters.some((adapter) => adapter.name !== wallet_adapter_mobile_1.SolanaMobileWalletAdapterWalletName &&
diff --git a/node_modules/@solana/wallet-adapter-react/lib/esm/getEnvironment.js b/node_modules/@solana/wallet-adapter-react/lib/esm/getEnvironment.js
index 2e8887e..475d3ec 100644
--- a/node_modules/@solana/wallet-adapter-react/lib/esm/getEnvironment.js
+++ b/node_modules/@solana/wallet-adapter-react/lib/esm/getEnvironment.js
@@ -6,7 +6,7 @@ export var Environment;
     Environment[Environment["MOBILE_WEB"] = 1] = "MOBILE_WEB";
 })(Environment || (Environment = {}));
 function isWebView(userAgentString) {
-    return /(WebView|Version\/.+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+)|; wv\).+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+))/i.test(userAgentString);
+    return /(Telegram|WebView|Version\/.+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+)|; wv\).+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+))/i.test(userAgentString);
 }
 export default function getEnvironment({ adapters, userAgentString }) {
     if (adapters.some((adapter) => adapter.name !== SolanaMobileWalletAdapterWalletName &&
diff --git a/node_modules/@solana/wallet-adapter-react/src/getEnvironment.ts b/node_modules/@solana/wallet-adapter-react/src/getEnvironment.ts
index acba961..72330ae 100644
--- a/node_modules/@solana/wallet-adapter-react/src/getEnvironment.ts
+++ b/node_modules/@solana/wallet-adapter-react/src/getEnvironment.ts
@@ -12,7 +12,7 @@ type Config = Readonly<{
 }>;
 
 function isWebView(userAgentString: string) {
-    return /(WebView|Version\/.+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+)|; wv\).+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+))/i.test(
+    return /(Telegram|WebView|Version\/.+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+)|; wv\).+(Chrome)\/(\d+)\.(\d+)\.(\d+)\.(\d+))/i.test(
         userAgentString
     );
 }
